extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java

```diff
@@ -40,37 +28,32 @@ public static FlywayDataSourceRuntimeConfig defaultConfig() {
      * <li>This schema will be the default for the database connection (provided the database supports this concept).</li>
      * </ul>
      */
-    @ConfigItem
-    public Optional<String> defaultSchema = Optional.empty();
+    Optional<String> defaultSchema();
 
     /**
      * The JDBC URL that Flyway uses to connect to the database.
      * Falls back to the datasource URL if not specified.
      */
-    @ConfigItem
-    public Optional<String> jdbcUrl = Optional.empty();
+    Optional<String> jdbcUrl();
 
     /**
      * The username that Flyway uses to connect to the database.
      * If no specific JDBC URL is configured, falls back to the datasource username if not specified.
      */
-    @ConfigItem
-    public Optional<String> username = Optional.empty();
+    Optional<String> username();
 
     /**
      * The password that Flyway uses to connect to the database.
      * If no specific JDBC URL is configured, falls back to the datasource password if not specified.
      */
-    @ConfigItem
-    public Optional<String> password = Optional.empty();
+    Optional<String> password();
 
     /**
      * Comma-separated case-sensitive list of schemas managed by Flyway.
      * The first schema in the list will be automatically set as the default one during the migration.
      * It will also be the one containing the schema history table.
      */
-    @ConfigItem
-    public Optional<List<String>> schemas = Optional.empty();
+    Optional<String[]> schemas();
```

Comment:
Why this change of types?

Reply:
Because `FluentConfiguration` expects a `String[]`.

See https://flywaydb.org/documentation/usage/api/javadoc/org/flywaydb/core/api/configuration/FluentConfiguration.html#schemas(java.lang.String...)

Reply:
Sure, but we don't really use `String[]` so I find it odd

Reply:
I reverted to `Optional<List<String>>` to avoid any unintentional type change